mne/preprocessing/ctps_.py

```diff
@@ -151,9 +151,9 @@ def _prob_kuiper(d, n_eff, dtype='f8'):
     j2 = (np.arange(n_points) + 1) ** 2
     j2 = j2.repeat(n_time_slices).reshape(n_points, n_time_slices)
     fact = 4. * j2 * l2 - 1.
-    expo = np.exp(-2. * j2 * l2)
+    expo = np.exp(-2. * j2 * l2, dtype='float128')
     term = 2. * fact * expo
-    pk = term.sum(axis=0, dtype=dtype)
+    pk = term.sum(axis=0)
```

Comment:
you should use https://docs.scipy.org/doc/scipy-0.19.1/reference/generated/scipy.misc.logsumexp.html

to increase numerical accuracy rather than casting to float128

Reply:
@agramfort : The issue of CTPS precision is further discussed in another pull request named 'ctps_precision'